### ISSUE TITLE
calculate co2-levels saved by the extension

### ISSUE THREAD

**Issue by SustDig** [uBlockOrigin/uBlock-issues#781]
## Description 
on average uBlock-usage, each time you open the uBlock-info, an small page is shown, enabling you to disable incoming and outgoing transfers to specific internet-addresses. this is great. 
wouldnt it be nice if **an additional info there** would calculate **how much co2 were saved** through this? this is not a trivial thing, sure, even if it sounds that way.  a current trend in european computing is to quantify exactly an average co2-footprint-scenario of each everyday-online-things (streaming, social-networks) a.s.o. - why not do so for tracking-/ad-networks?

- a first hint could be studies on the average co2-footprint of each mail you sent (studies by greenpeace and others)
- another hint is a already existing firefox-extension: https://github.com/supertanuki/Carbonalyser
- if the first real studies here should appear - and uBlock-development could be avantgarde here, not only in the problem-constitution, but also in the solution - it would be highly expectable that real figures from the sites itself could appear and could be incorporated (in the long run even leading to some form of low-carbon-footprint-certificates for ad-/tracking-sites - and you could lead their development too).
- nice next steps could be imagined: also simply calculated daily, weekly, mothly, yearly individual effects of the current you-do-it-for-me-attitude in coding via incorporating all this bunch of google-analytics, external fonts, external javascript-libraries in nearly all internet-pages
- while being one of the top browsing-extensions currently, you could gain leadership in an not-yet-really-existing field of greenbrowsing - or you could even be the first coining the term.

## Steps to Reproduce
   1. use the extension, co2-footprint of browsing is not shown

## Actual result:
co2-footprint of blocked Ads and trackers not shown.

## Expected result:
co2-footprint of blocked Ads and trackers should be calculated here

## Reproduces how often: 
infinite until new feature

## Brave version (brave://version info)
git latest

## Version/Channel Information:
same

## Other Additional Information:
none

## Miscellaneous Information:
none

**Comment by gorhill** [comment]
Declined.

This is essentially a duplicate of <https://github.com/uBlock-LLC/uBlock/issues/31>: you can't know what would be loaded/parsed/executed resources without actually loading/parsing/executing resources.

**Comment by SustDig** [comment]
>gorhill: thats why i linked https://github.com/supertanuki/Carbonalyser giving good average values for a first try. the enhancement i suggested is currently more a political problem of technology (which should be reduced and which effects cannot be quantified completely exact currently but can be approximated to measure the problem) then an already fixed technical solution - like adblockers were when they were not invented / implemented technically yet. An Idea to think about (which could significantly broaden the usage-base of your extension in the future and bring in a totaly new user-community).

**Comment by SustDig** [comment]
the thing completely easy: take a look at https://www.websitecarbon.com/ this information could be included.

**Comment by Sophivorus** [comment]
@gorhill Hi! An easy way to implement this functionality would be:

Hard-code a brute average of the amount of energy saved per blocked request and then simply multiply by the number of requests blocked. For example, according to https://onlinelibrary.wiley.com/doi/epdf/10.1111/jiec.12630 data transfer consumes **0.06 kWh/GB**, multiplied by the **low estimate of 200 bytes per request** (according to https://stackoverflow.com/questions/5358109) and multiplied again by the **761,837 requests** uBlock Origin has blocked for me, equals 32911 Joules saved (according to https://www.google.com/search?q=761837*200+bytes*0.06+kWh%2FGB) Thanks!

Of course these are first estimates and more refinements and research would be needed, **but I'd be happy to do that**. Also, an equivalence between Joules saved and some more familiar measure of energy would be welcome.

Lastly, if there're stats on the total number of requests every blocked by uBlock Origin, it'd be possible to display the total energy saved at the website and extension description.

I really hope you give some thought to these ideas. In any case, thanks for all the energy saved!

**Comment by gorhill** [comment]
I consider this to be feature creep and best left to a separate extension -- another extension can listen to which network requests were blocked (by whichever extension(s)) and make of this whatever it wants.
